executor/join_result_generators.go

```diff
@@ -33,41 +35,109 @@ var (
 
 // joinResultGenerator is used to generate join results according the join type, see every implementor for detailed information.
 type joinResultGenerator interface {
+	// outerIdx returns the child index of outer table.
+	outerIdx() int
+	// initDefaultChunkInner converts default inner rows stored in a Datum slice to a chunk.Row.
+	initDefaultChunkInner(innerTypes []*types.FieldType)
+
 	// emitMatchedInners should be called when key in outer row is equal to key in every inner row.
 	// Reutrn true if outer row can be joined with any input inner row.
-	emitMatchedInners(outer Row, inners []Row, resultBuffer []Row) ([]Row, bool, error)
+	emitMatchedInners(outer Row, inners []Row, resultBuffer []Row) ([]Row, error)
+	emitMatchedInnersToChunk(outer chunk.Row, inners []chunk.Row, chk *chunk.Chunk) error
+
 	// emitUnMatchedOuter should be called when outer row is not matched to any inner row.
 	emitUnMatchedOuter(outer Row, resultBuffer []Row) []Row
+	emitUnMatchedOuterToChunk(outer chunk.Row, chk *chunk.Chunk)
```

Comment:
Can we remove all the `outer` methods in the interface?

Reply:
I think the current method naming is more specific and straightforward, users can guess their actions without any hesitation and thinking.